### DISCUSSION TITLE
Is the use of make due to the limitations of rust?

### DISCUSSION THREAD

**Discussion by JustinMBrown** [uutils/coreutils#4116] [General]
And if so, what could Rust do to allow y'all to not have to use make? Add features to cargo? Create a new build tool that have the same features as make?

Sorry if it's a dumb question, I just thought it was weird to use gnu software when trying to replace gnu software.

**Comment by sylvestre** [comment]
The goal of this project, to me at least, isn't to replace GNU software. I am using it everyday for many things and I love these projects.
The goal is to provide a Rust implementation of a core piece of infrastructure.

And if you look at the source of https://github.com/uutils/coreutils/blob/main/GNUmakefile, it is mostly doing calls to cargo.
it could be a script or something else.

**Comment by oech3** [comment]
I think installing this project without `make` is possible if `uudoc` got a feature to create symlinks of `coreutils` and create dirs before installing manpages. This project already have crates to do that.

But I think we still need `make` or some script to filter-out some features.
